config.dev.toml

```diff
@@ -290,6 +290,17 @@ SingleFlightType = "memory"
         # TODO(marwan): enable multiple endpoints for redis clusters.
         # Env override: ATHENS_REDIS_ENDPOINT
         Endpoint = "127.0.0.1:6379"
+    [SingleFlight.RedisSentinel]
```

Comment:
Looks like the options here are the same except for `MasterName` -- would it be possible to just re-use SingleFlight.Redis and add the MasterName field there? If MasterName exists, then use `NewFailoverClient` otherwise just use `NewClient` 

In the documentation here, we'd mention this logic. 

This would keep our large configuration file from becoming larger. 

But if there's a problem with this definitely let me know. 

Thanks

Reply:
I see `Endpoint` is `Endpoints`, maybe we can combine those and make it backwards incompatible..it would get rid of the TODO I put there :)

Reply:
When it comes to accepting multiple endpoints on a non-sentinel redis setup, what would the expected behavior be? Looking at `go-redis`' implementation, I don't think having multiple endpoints would satisfy the need of having a consistent lock at least based on the implementation of `redis.ClusterClient`

Reply:
Coming from a docs / usability perspective, I think that we should keep sentinel config separate from plain old redis config. Even though configs are similar, the underlying systems are different, so I think we should reflect that as much as possible.

Reply:
@arschles's point is spot on, too. Regardless, since I'm tooling around in here, it may be beneficial either here, or in a quick-follow up to resolve @marwan-at-work's `TODO`...but there's a pretty big implementation detail that needs to be sussed out, as mentioned in my comment.

Reply:
I'm happy either way, but we should just be mindful of adding more configs because it will very hard to remove them :) 

I'm not familiar with Redis Sentinel, @arschles how is it a different system? It looks like the client object is exactly the same.

Reply:
@marwan-at-work I bet @twexler can fill in more info, but last I used it, sentinel was a cluster of servers that cooperate. You can then issue a write or read to any of them, and they will load balance and failover.

Your point about removing configs is very well taken, and I am reconsidering my point on that. I wonder if it's understandable enough to merge the configs & keep backward compatibility like you said, and then document how to use sentinel _really_ well 😄